azure-pipelines.yml

```diff
@@ -66,36 +79,22 @@ jobs:
       DOCKERHUB_PASSWORD: $(DockerPassword)
       GCP_SERVICE_ACCOUNT_KEY: $(GcpServiceAccountKey)
 
-- job: generate_toolchains_linux
-  dependsOn: "build_container_linux"
+- job: generate_toolchains_windows
+  dependsOn: "build_container_windows"
   pool:
-    vmImage: 'ubuntu-18.04'
+    vmImage: 'windows-latest'
   steps:
-  - bash: |
-      sudo curl -sSL -o /usr/local/bin/bazel https://github.com/bazelbuild/bazelisk/releases/download/v1.4.0/bazelisk-linux-amd64
-      sudo chmod +x /usr/local/bin/bazel
-    displayName: Install Bazelisk
-
-  - task: InstallSSHKey@0
-    inputs:
-      hostName: "github.com ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEAq2A7hRGmdnm9tUDbO9IDSwBK6TbQa+PXYPCPy6rbTrTtw7PHkccKrpp0yVhp5HdEIcKr6pLlVDBfOLX9QUsyCOV0wzfjIJNlGEYsdlLJizHhbn2mUjvSAHQqZETYP81eFzLQNnPHt4EVVUh7VfDESU84KezmD5QlWpXLmvU31/yMf+Se8xhHTvKSCZIFImWwoG6mbUoWf9nzpIoaSjB+weqqUUmpaaasXVal72J+UX2B+2RPW3RcT0eOzQgqlJL3RKrTJvdsjE3JEAvGq3lGHSZXy28G3skua2SmVi/w4yCE6gbODqnTWlg7+wC604ydGXA8VJiS5ap43JXiUFFAaQ=="
-      sshPublicKey: "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCns5kFJ6w7c/zH/FYBNvSpqhpSL147fizsru+PYwaw36RMCkJ4wqHVqusq6zF6K8SPloA4AlzPWx50i2NhRj8NEKgg3IkO35uFciEBcTwqr5+o+e96CvryNwGDZfavgUfrMwTQBYWdM7aaBoWUk31zmUixtlalgBEC1JGdEFxYpN3AmqN50mbu9ZvV8IEv8/AInqvqkS+dIzE9EcZQZQ98uDRU99S1YDCcvZO6xidGZz42AbS8CJ6idrreBoqy/eXmgio4YjGGFRnqr1tdXG5GKWSxfcrSC/gE8OzpFTJ1WOWH+SdZCptev8er6MBoFgRNkDhRckzpsAjhIYxSIAaL"
-      sshPassphrase: "$(SshKeyPassphrase)"
-      sshKeySecureFile: id_rsa_envoy_ci
-    condition: ne(variables['Build.Reason'], 'PullRequest')
-
-  - bash: |
-      toolchains/regenerate.sh
-    env:
-      COMMIT_TOOLCHAINS: "true"
-      SOURCE_BRANCH: $(Build.SourceBranch)
-    displayName: Generate toolchains
-
+  - template: .azure-pipelines/generate_toolchains.yml
+    parameters:
+      bazeliskInstallCommandPrefix: ""
+      bazeliskInstallPath: "/usr/bin/bazel.exe"
+      bazeliskReleaseFilename: bazelisk-windows-amd64.exe
+      gcrImageName: "envoy-build-windows"
+      osFamily: "windows"
+  # Publish build artifacts in last step of the pipeline
```

Comment:
@lizan not sure if this is the purpose of this artifact publish task, it seems like the intent is to run as the last step of the pipeline after everything has been committed, does this look right to you? (only doing it once after the last job)

Reply:
@lizan also, can you clarify what these published artifacts are for/where are they used? is this task still necessary?

Reply:
No it isn't necessary. I was used to use this tar to test generated toolchains from CI.